File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        int len = s.length();
        boolean isValid = true;
        for(int i=0;i<len;i++){
            char ch = s.charAt(i);
            if(!st.isEmpty() && (ch == ')' || ch == '}' || ch == ']')){
                if(ch == ')' && st.peek() != '('){
                    isValid = false;
                    break;
                }
                else if(ch == '}' && st.peek() != '{'){
                    isValid = false;
                    break;
                }
                else if(ch == ']' && st.peek() != '['){
                    isValid = false;
                    break;
                }
                else{
                    st.pop();
                }
            }
            else{
                st.push(ch);
            }
        }
        return st.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for(char ch : s){
            if(ch == '(' || ch == '{' || ch == '[') st.push(ch);
            else{
                if(st.empty()) return false;
                char  x = st.top();
                if((ch == ')' && x == '(') || (ch == '}' && x == '{') || (ch == ']' && x == '[')){
                    st.pop();
                }
                else return false;
            }
        }
        return st.empty();
    }
};